### ISSUE TITLE
Tversky loss function for image segmentation using 3D fully convolutional deep networks

### ISSUE THREAD

**Issue by Yagami360** [Yagami360/MachineLearning-Papers_Survey#67]
## 0. 論文情報・リンク
- 論文リンク：https://arxiv.org/abs/1706.05721
- 公開日時：2017/06/18
- 被引用数（記事作成時点）：78 件
- 実装コード：
- Publication : 

## 1. どんなもの？
- セグメンテーションセグメンテーションにおいて、学習用データのクラス間不均衡（あるラベルの領域が圧倒的に広く別のラベル領域が値小さい）がある場合にもうまく学習出来るように、dice loss を一般化した Tversky loss を提案し、このようなケースでのセグメンテーション品質を向上させている。

## 2. 先行研究と比べてどこがすごいの？
- セグメンテーションセグメンテーションにおいて、学習用データのクラス間不均衡（あるラベルの領域が圧倒的に広く別のラベル領域が値小さい）がある場合には、損失関数は局所的最適解の陥るなどの理由により、セグメンテーション画像の品質が悪化してしまうことが知られている。
    クラス間不均衡を解決する方法として、dice loss があるが、本手法ではこの dice loss を一般化した Tversky loss を採用することでクラス間不均衡により対処できるようにし、このようなケースでのセグメンテーション品質を向上させている。

## 3. 技術や手法の"キモ"はどこにある？
- Tversky loss
	セグメンテーションセグメンテーションにおいて、学習用データのクラス間不均衡（あるラベルの領域が圧倒的に広く別のラベル領域が値小さい）がある場合には、損失関数は局所的最適解の陥るなどの理由により、セグメンテーション画像の品質が悪化してしまうことが知られている。
	これらのクラス間不均衡を解決する方法としては、voxelwise methods や re-weighting（再重み付け）などの方法があるが、どちらの方法も一部のクラスに overfit しやすいといった問題や学習初期状態鋭敏性などの問題がある。
	
	クラス間不均衡を解決する別の方法としては、dice loss がある。
	この dice loss は、precision（適合率）と recell（再現率）の調和平均で定義されているので、偽陽性(FP)と偽陰性(FN)を等しく重み付けする。これにより、一部のクラスに overfit しやすいといった問題や学習初期状態鋭敏性などの問題が回避されている。
    本手法では、この dice 係数を一般化したTversky 指数に基づく Tversky loss を採用し、precision（適合率）と recell（再現率）のよりよい調和を行う。これにより、クラス間不均衡により対処できるようになる。

    Tversky 指数は、以下の式で定義される。
    <image src="https://user-images.githubusercontent.com/25688193/85403729-d4ea4200-b598-11ea-908c-8d16605639bb.png" width="470">

    この Tversky 指数に基づく Tversky loss は、以下の式のように定義される。
    <image src="https://user-images.githubusercontent.com/25688193/85403779-e9c6d580-b598-11ea-93e0-a5a60ab3f9f0.png" width="575">
	先に述べたように、dice 係数に基づく Dice loss による学習では、偽陽性(FP)と偽陰性(FN)を等しく重み付けしながら学習が行われる。一方 Tversky loss では、上式を見てわかるように、偽陽性(FP)より偽陰性(FN)を強く重み付けすることが出来る。
これにより、小さなクラス（領域が小さいクラス）により強く反応するように出来るので、結果としてクラス間不均衡に対処出来るようになる。

- 【補足】Dice loss
	２つの集合の類似度を評価する指標の１つに Dice 係数がある。
	これは、以下の式のように２つの集合 A, B の平均要素数と共通要素数の割合を示した指標になっており、Dice 係数値が大きいほど２つの集合の類似度が高くなる。（値の範囲は 0 ~ 1）
    <image src="https://user-images.githubusercontent.com/25688193/85403852-05ca7700-b599-11ea-9276-b92250733b3e.png" width="200">

	セマンティクスセグメンテーションモデルにおいては、この Dice 係数を元にした dice loss が広く採用されている。
	※ dice loss を最初に提案した論文は「V-Net: Fully Convolutional Neural Networks for Volumetric Medical Image Segmentation」
	
	これは以下の式のように、２つの集合 A, B をそれぞれ推論マスク画像と正解マスク画像とし、画像中の全ピクセルの和とした損失関数であり、推論マスク画像と正解マスク画像の類似度が高くなるにつれて損失関数値が小さくなるように設計されている。
    <image src="https://user-images.githubusercontent.com/25688193/85403919-209ceb80-b599-11ea-9925-f6592a9ce70c.png" width="525">

## 4. どうやって有効だと検証した？
- Dice loss と Tversky loss との定性比較
    <image src="https://user-images.githubusercontent.com/25688193/85404041-56da6b00-b599-11ea-9d3e-0f4bff78c695.png" width="800">
	α=β=0.5 のときは、Dice loss での結果と同じになる。
	Tversky loss で、α<β としたほうが品質が向上していることが見て取れる。（ベストは α=0.3,β=0.7 のとき）

- Dice loss と Tversky loss との定量比較
    <image src="https://user-images.githubusercontent.com/25688193/85404475-f0a21800-b599-11ea-9ded-198cf000ba66.png" width="480">
	Tversky loss で、α<β としたほうが特に F2スコア（Precision と recall の調和平均）で品質が向上していることが見て取れる。（ベストは α=0.3,β=0.7 のとき）

## 5. 議論はあるか？
- xxx

## 6. 次に読むべき論文はあるか？
- V-Net: Fully Convolutional Neural Networks for Volumetric Medical Image Segmentation
- A Novel Focal Tversky loss function with improved Attention U-Net for lesion segmentation
- Boundary loss for highly unbalanced segmentation

## 7. 参考文献
- xxx
